code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -66,31 +81,45 @@
 	return ..()
 
 /obj/structure/closet/update_icon()
```

Comment:
```suggestion
/obj/structure/closet/update_icon_state()
```

You split update icon into update_overlays already, might as well do this too.

Recommend changing every call in this file from update_icon to `update_appearance` as well